src/snr_measurements.py

```diff
@@ -305,8 +306,102 @@ def check_xt(self):
             resp = cross_talk < self.engine_props['requested_xt'][self.spectrum_props.modulation]
 
         return resp, cross_talk
+    
+    # TODO: update the method based on external resources
+    def check_snr_ext(self, path_index):
+        """
+        Checks the SNR on a single request using the external resources.
+
+        :return: Whether the SNR threshold can be met and SNR value.
+        :rtype: tuple
+        """
+        mod_format_mapping = {
```

Comment:
If you want, I think there's a snr_args.py script you could put all of your arguments, up to you